fslock_windows.go

```diff
@@ -0,0 +1,12 @@
+package fslock
+
+import (
+	"errors"
+	"strings"
+
+	"golang.org/x/sys/windows"
+)
+
+func lockedByOthers(err error) bool {
+	return errors.Is(err, windows.ERROR_SHARING_VIOLATION) || strings.Contains(err.Error(), "being used by another process")
```

Comment:
```suggestion
	return errors.Is(err, windows.ERROR_SHARING_VIOLATION) || strings.Contains(err.Error(), windows.ERROR_SHARING_VIOLATION.Error())
```

Reply:
^this pattern should probably be repeated for the other platforms as well.
That is, in the `_unix` file, don't use the equality operator to check syscall error values, instead use `errors.Is` with them; and call their `Error` method too.
If the Unix error value on the lefthand of the OR isn't already matched up with the one we expect to receive from `lock.Lock` we can update it.

As is, I don't think the lefthand condition could ever be true, since `Lock` (currently) returns a string error, never a `syscall.Errno`.